### PR TITLE
Add: The redux substate for the site topic picking step in the improved onboarding signup flow.

### PR DESCRIPTION
**Pull request by southp**
#### Changes proposed in this Pull Request

This PR adds the redux state part of the site topic picking step in the improved onboarding signup flow. The UI part can be found in https://github.com/Automattic/wp-calypso/pull/28146.

#### Testing instructions
Run `npm test-client site-topic`

### CODE REVIEW

File: client/state/signup/steps/reducer.js [hunk 1]
```diff
@@ -14,6 +15,7 @@ import survey from './survey/reducer';
 export default combineReducers( {
 	designType,
 	siteTitle,
+	siteTopic,
```
Comment (ramonjd) [inline client/state/signup/steps/reducer.js]: Should we add it to the bundle later? Assuming this is going out first?

Comment (southp) [reply client/state/signup/steps/reducer.js]: It should be safe, because nothing really uses this substate tree.

Comment (ramonjd) [reply client/state/signup/steps/reducer.js]: It was mainly to avoid extra bytes to the bundle where don't yet need to.

Comment (southp) [reply client/state/signup/steps/reducer.js]: Ohhhhh. That makes sense. I'll remove it from here before deploying it.

**Pull request merged**